internal/app/frontend/frontend_service_test.go

```diff
@@ -88,6 +88,62 @@ func TestDoCreateTickets(t *testing.T) {
 	}
 }
 
+func TestDoCreateBackfill(t *testing.T) {
+	cfg := viper.New()
+
+	tests := []struct {
+		description string
+		preAction   func(cancel context.CancelFunc)
+		backfill    *pb.Backfill
+		wantCode    codes.Code
+	}{
+		{
+			description: "expect error with canceled context",
+			preAction:   func(cancel context.CancelFunc) { cancel() },
+			backfill: &pb.Backfill{
+				SearchFields: &pb.SearchFields{
+					DoubleArgs: map[string]float64{
+						"test-arg": 1,
+					},
+				},
+			},
+			wantCode: codes.Unavailable,
+		},
+		{
+			description: "expect normal return with default context",
+			preAction:   func(_ context.CancelFunc) {},
+			backfill: &pb.Backfill{
+				SearchFields: &pb.SearchFields{
+					DoubleArgs: map[string]float64{
+						"test-arg": 1,
+					},
+				},
+			},
+			wantCode: codes.OK,
+		},
+	}
+
+	for _, test := range tests {
+		test := test
+		t.Run(test.description, func(t *testing.T) {
+			store, closer := statestoreTesting.NewStoreServiceForTesting(t, cfg)
+			defer closer()
+
+			ctx, cancel := context.WithCancel(utilTesting.NewContext(t))
+			test.preAction(cancel)
+
+			res, err := doCreateBackfill(ctx, &pb.CreateBackfillRequest{Backfill: test.backfill}, store)
```

Comment:
require.NotNil(t, res)

Reply:
Why do you test doCreateBackfill instead of CreateBackfill?

Reply:
I am testing real handlers as part of E2E test.

Reply:
This is a unit test, it has nothing to do with e2e one. _doCreateBackfill_ is not an entry point to the current package. Please, test exported methods only, as you do in TestDoGetBackfill.

Reply:
First of all we follow the same pattern for `TestDoCreateTicket()`, second Scott asked me to use E2E to cover those handlers:
https://github.com/googleforgames/open-match/pull/1279/files#r528961563

Reply:
If there is the same approach in TestDoCreateTicket it doesn't mean that it is done right. You should not rely on any other tests (e2e, etc) while working on unit tests. Here you **ignore** everything which comes before _doCreateBackfill_.

Reply:
@Laremere do you mean that we should test public API in e2e tests, rather than unit ones?
Because at this moment I don't see a clear boundary between tickets e2e and tickets unit tests, for instance.
If so - there is no point to add and support tests in _frontend_ package right now.

Reply:
With _TestDoCreateTickets_ half of a method is tested in frontend package, another part (validations) in e2e, which is a bit confusing.